frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx

```diff
@@ -55,112 +43,62 @@ const DisplayAssessmentsPage = (): React.ReactElement => {
-    { label: "Grade", setState: setGrades, options: gradeOptions },
-    { label: "Type", setState: setTestTypes, options: testTypeOptions },
-    { label: "Country", setState: setCountries, options: countryOptions },
-    { label: "Region", setState: setRegions, options: regionOptions },
-  ];
+  const [filterOptions, setFilterOptions] = React.useState<FilterProp[]>([
+    { label: "Grade", setState: setGrades, options: [] },
+    { label: "Type", setState: setTestTypes, options: [] },
+    { label: "Country", setState: setCountries, options: [] },
+    { label: "Region", setState: setRegions, options: [] },
+  ]);
 
   const { loading, error, data } = useQuery(GET_ALL_TESTS, {
     fetchPolicy: "cache-and-network",
+    onCompleted: () => {
+      const {
+        gradeOptions,
+        testTypeOptions,
+        countryOptions,
+        regionOptions,
+      } = assessmentFilterOptions(data.tests);
+
+      setFilterOptions((prev) => {
+        return [
+          {
+            ...prev[0],
+            options: gradeOptions,
+          },
+          {
+            ...prev[1],
+            options: testTypeOptions,
+          },
+          {
+            ...prev[2],
+            options: countryOptions,
+          },
+          {
+            ...prev[3],
+            options: regionOptions,
+          },
+        ];
+      });
+    },
   });
 
-  const filteredAssessements = React.useMemo(() => {
+  const filteredAssessments = React.useMemo(() => {
     if (!data) return [];
-
-    let filteredTests: AssessmentProperties[] = data.tests as AssessmentProperties[];
-    const filterProps = [grades, testTypes, countries, regions, status];
-
-    if (filteredTests.length) {
+    if (data.tests.length) {
       setEmpty(false);
     }
 
-    filterProps.forEach((property, i) => {
-      filteredTests = filteredTests.filter(
-        (assessment: AssessmentProperties) => {
-          const assessmentProperties = [
-            assessment.grade,
-            assessment.assessmentType,
-            assessment.curriculumCountry,
-            assessment.curriculumRegion,
-            assessment.status,
-          ];
-          if (property.length === 0) {
-            return true;
-          }
-          return property.includes(assessmentProperties[i]);
-        },
-      );
-    });
-
-    return filteredTests;
+    const filterProps = [grades, testTypes, countries, regions, status];
+    return filterAssessments(data.tests, filterProps);
   }, [data, grades, testTypes, countries, regions, status]);
 
   const searchedAssessements = React.useMemo(() => {
-    let filteredTests = filteredAssessements;
-    if (search) {
-      filteredTests = filteredTests.filter(
-        (assessment: AssessmentProperties) =>
-          assessment.name.toLowerCase().includes(search.toLowerCase()) ||
-          removeUnderscore(assessment.grade)
-            .toLowerCase()
-            .includes(search.toLowerCase()) ||
-          getFirstNumber(assessment.grade)
```

Comment:
i'm not sure why this was needed in the first place?? so i removed it. but lmk if we need it